### ISSUE TITLE
onClickCapture event handler prevents slide onClick events from triggering (when dragging is disabled)

### ISSUE THREAD

**Issue by Francesco-Lanciana** [FormidableLabs/nuka-carousel#399]
Bug report

**What is the current behavior?**
When children of carousel have an onClick event handler it won't be triggered unless you have set the dragging option to true.

**If the current behavior is a bug, please provide the steps to reproduce.**

Clicking the div in the Carousel rendered by the following React component will not trigger the console.log:
```
import React from "react";
import Carousel from 'nuka-carousel';

const BrokenCarousel = () => (
    <Carousel dragging={false}>
        <div onClick={() => { console.log("What does it matter, I won't appear anyway") }} style={{width: "100px", height: "100px", background: "blue"}}/>  
    </Carousel>
)

export default BrokenCarousel;
```

If you trigger a touch event (by changing the screen to a phone) then you will see the log appear in the console. The only way to circumvent this (that I have discovered) is to set draggable to true.

**What is the expected behavior?**
The onClick handler should be invoked regardless of what draggable is set to.

Either the onClickCapture should be removed to allow the onClick handler of any children of the carousel to be invoked before the onClick handler of the carousel itself.
OR the event propogation should not be stopped within the handleClick method
OR something else that prevents this problem.

**Other relevant information:**
The reason it works when dragging is set to true is because of the following:
* onMouseUp is invoked before onClick.
* onMouseUp invokes handleSwipe which sets this.clickSafe to false in the event that there was no movement (which there wasn't because we only clicked). 
* handleClick now gets invoked but this.clickSafe is set to false so propagation of the current event is not halted. 

The following commit triggered this behaviour:

[fix extra click event triggered when dragging slide](https://github.com/FormidableLabs/nuka-carousel/commit/860f2db54f82aa010edefa4769748d485eeab7ef#diff-1fdf421c05c1140f6d71444ea2b27638)

**Comment by minipai** [comment]
👍

**Comment by yantonsoup** [comment]
Bump on this, causing lots of issues over here.

**Comment by benjamminj** [comment]
Just came up on this issue today, would love to do a fix if anyone wants to point me in the right direction.

Until we get this fixed, [here's an CodeSandbox](https://codesandbox.io/s/p49nmjwl7) of my temporary workaround. The gist is basically is leave the carousel with `draggable={true}` mode and intercept / cancel the mouse-move events on my own.

**Comment by lukeliasi** [comment]
As a temporary fix you can change your `onClick` to `onMouseDown` and it should work.

**Comment by varenya** [comment]
the temporary fix won't help with anchor tags what can one do for that?

**Comment by weizijian101** [comment]
pointer-events: all

**Comment by jflayhart** [comment]
What do y'all think about adding an "onItemTap" event as a nuka-carousel prop? If I get some time, I'd be happy to make a PR.

**Comment by jflayhart** [comment]
Another thing worth mentioning as that even with draggable set to TRUE (default), this does not work for us as we have other custom elements that fire individual onClick events. Therefore, when ANY onClick event fires within the carousel, it only takes the last event that propagated.

**Comment by jflayhart** [comment]
Welp... just as a FYI, I had to fork and move away from the maintainers of this library: https://github.com/jflayhart/nuka-carousel/commit/384b2563ad880b8c7038cdc6709f41154a3b5f55

I understand this won't work for this project because it would break all the users who depend on this library for stopping propagation and preventing click events, but the fact is I believe this lib should NOT be responsible for doing that. It should be up to the installer of this library to account for how they want to handle said click events. With my fork, this library works great for us!

**Comment by jflayhart** [comment]
@sarmeyer https://github.com/FormidableLabs/nuka-carousel/pull/423

**Comment by sarmeyer** [comment]
fixed in #423
